### PR TITLE
JAMES-3904 Support LDAPS

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/apps/distributed-app/docs/modules/ROOT/pages/configure/usersrepository.adoc [hunk 1]
```diff
@@ -62,4 +62,14 @@ Example:
...
-....
\ No newline at end of file
+....
+
+SSL can be enabled by using `ldaps` scheme. `trustAllCerts` option can be used to trust all LDAP client certificates.
+
+Example:
+
+....
+<repository name="LocalUsers" class="org.apache.james.user.ldap.ReadOnlyUsersLDAPRepository" ldapHost="ldaps://myldapserver:389"
```
Comment (quantranhong1999) [inline server/apps/distributed-app/docs/modules/ROOT/pages/configure/usersrepository.adoc:72]: Shouldn't the SSL port of LDAP be 636?

File: server/data/data-ldap/src/test/java/org/apache/james/user/ldap/LdapGenericContainer.java [hunk 2]
```diff
@@ -121,4 +121,11 @@ public String getLdapHost() {
                 ":" +
                 LdapGenericContainer.DEFAULT_LDAP_PORT;
     }
+
+    public String getLdapsHost() {
+        return "ldaps://" +
+                container.getContainerIp() +
+                ":" +
+                LdapGenericContainer.DEFAULT_LDAP_PORT;
```
Comment (quantranhong1999) [inline server/data/data-ldap/src/test/java/org/apache/james/user/ldap/LdapGenericContainer.java:129]: LDAP_SSL_PORT 636?

File: src/site/xdoc/server/config-users.xml [hunk 3]
```diff
@@ -97,6 +97,16 @@
 &lt;repository name="LocalUsers" class="org.apache.james.user.ldap.ReadOnlyUsersLDAPRepository" ldapHost="ldap://myldapserver:389"
     principal="uid=ldapUser,ou=system" credentials="password" userBase="ou=People,o=myorg.com,ou=system" userIdAttribute="uid"/&gt;</source>
 
+
+        <p>SSL can be enabled by using <code>ldaps</code> scheme. <code>trustAllCerts</code> option can be used to trust all LDAP client certificates.</p>
+
+        Example:
+
+        <source>
+ &lt;repository name="LocalUsers" class="org.apache.james.user.ldap.ReadOnlyUsersLDAPRepository" ldapHost="ldaps://myldapserver:389"
```
Comment (quantranhong1999) [inline src/site/xdoc/server/config-users.xml:106]:
```suggestion
 &lt;repository name="LocalUsers" class="org.apache.james.user.ldap.ReadOnlyUsersLDAPRepository" ldapHost="ldaps://myldapserver:636"
```

**Review by Arsnael — APPROVED** [review]

File: server/apps/distributed-app/docs/modules/ROOT/pages/configure/usersrepository.adoc [hunk 4]
```diff
@@ -62,4 +62,14 @@ Example:
 ....
 <repository name="LocalUsers" class="org.apache.james.user.ldap.ReadOnlyUsersLDAPRepository" ldapHost="ldap://myldapserver:389"
     principal="uid=ldapUser,ou=system" credentials="password" userBase="ou=People,o=myorg.com,ou=system" userIdAttribute="uid"/>;
-....
\ No newline at end of file
+....
+
+SSL can be enabled by using `ldaps` scheme. `trustAllCerts` option can be used to trust all LDAP client certificates.
```
Comment (quantranhong1999) [inline server/apps/distributed-app/docs/modules/ROOT/pages/configure/usersrepository.adoc:67]: Notate default value `false` for `trustAllCerts`?

**Comment by Arsnael** [comment]
```
04:00:43,824 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:check (check-style) on project james-server-data-ldap: You have 4 Checkstyle violations. -> [Help 1]
```

**Review by quantranhong1999 — COMMENTED** [review]
`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:check (check-style) on project james-server-data-ldap: You have 4 Checkstyle violations.`

**Review by Arsnael — APPROVED** [review]

**Pull request merged**